panels/user-accounts/um-password-dialog.c

```diff
@@ -219,10 +220,14 @@ accept_password_dialog (GtkButton        *button,
                         break;
 
                 case ACT_USER_PASSWORD_MODE_SET_AT_LOGIN:
-                        act_user_set_password_mode (um->user,  um->password_mode);
                         act_user_set_automatic_login (um->user, FALSE);
+                case ACT_USER_PASSWORD_MODE_NONE:
```

Comment:
I would use a break statement here, even if it's not strictly necessary and adds a bit more code, for readability and robustness to future modification, including possible upstream modification.